### DISCUSSION TITLE
How to Aspire with AddPooledDbContextFactory

### DISCUSSION THREAD

**Discussion by jkears** [microsoft/aspire#2752] [Q&A] [answered]
I have an existing downstream service that connects to EF Core SQL as per this configuration method that is called from it's program.cs startup and I am using KeyVault to obtain the SQL Connection string ...

```
public static IServiceCollection AddDbContext(this IServiceCollection services, IConfiguration configuration, SecretClient secretClient)
     {
         var primarySqlConfigKey = configuration["PrimarySQLConfigKey"];
         if (string.IsNullOrEmpty(primarySqlConfigKey))
         {
             primarySqlConfigKey = "PrimaryPortDomainServiceDbConn";
         }
         // Retrieve KeyVault secrets for initialization
         var primaryPortDomainServiceDbConnSecret = secretClient.GetSecret(primarySqlConfigKey);
         services.AddPooledDbContextFactory<AppDbContext>((_, o) => 
             o.UseSqlServer(primaryPortDomainServiceDbConnSecret.Value.Value,
                     b => 
                             b.MigrationsAssembly("NextWare.UI.Builder.EditServices.API")));
         return services;
     }
```
As can be seen, I am utilizing Pooled DB Context Factory via `AddPooledDbContextFactory`

In the service project I replaced the existing EntityFrameworkCore with this package reference ...

`<PackageReference Include="Aspire.Microsoft.EntityFrameworkCore.SqlServer" Version="8.0.0-preview.2.23619.3" />`

As I can't seem to call AddPooledDbContextFactory to set settings as per the following;  'Tracing = true', ... I instead added the following configuration within the service project as follows ...

```
 "Aspire": {
   "Microsoft": {
     "EntityFrameworkCore": {
       "SqlServer": {
         "DbContextPooling": true,
         "HealthChecks": true,
         "Tracing": true,
         "Metrics": true
       }
     }
   }
 }
```
While this runs as it did prior to Aspire, I do not see any tracing wrt calls made against the AppDbContext.

**Comment by davidfowl** [comment]
cc @sebastienros

**Comment by bdukes** [comment]
Same basic question, trying to call `UseQuerySplittingBehavior` when using `AddSqlServerDbContext` from `Aspire.Microsoft.EntityFrameworkCore.SqlServer` (version 8.1.0)

**Comment by sebastienros** [comment, marked as answer]
A new API has been added since this question was asked. You can now configure the DB context by yourself using `EnrichSqlServerDbContext`. See https://github.com/dotnet/aspire/blob/main/src/Components/Aspire.Microsoft.EntityFrameworkCore.SqlServer/README.md#usage-example

```c#
var connectionString = builder.Configuration.GetConnectionString("catalogdb");
builder.Services.AddDbContextPool<CatalogDbContext>(dbContextOptionsBuilder => dbContextOptionsBuilder.UseSqlServer(connectionString));
builder.EnrichSqlServerDbContext<CatalogDbContext>();
```
